zrml/prediction-markets/src/lib.rs

```diff
@@ -570,123 +545,64 @@ mod pallet {
             Ok(Some(T::WeightInfo::create_market().saturating_add(extra_weight)).into())
         }
 
-        /// This function combines the creation of a market, the buying of a complete set of
-        /// outcome assets, the deployment of the minimum amount of outcome assets and
-        /// the optional deployment of additional outcome asset.
+        /// Buy complete sets and deploy a pool with specified liquidity for a market.
         ///
         /// # Arguments
         ///
-        /// * `market_id`: Id of the market for that the pool should be created and populated.
-        /// * `amount_base_asset`: The amount of the base asset that should be deployed.
-        /// * `amount_outcome_assets`: A vector containing the amount of each outcome asset that should be
-        ///     deployed. The highest value will be used to buy a complete set, i.e. every outcome
-        ///     asset will be bought in quantities specified by the highest value in this vector.
-        ///     Any value that is lower than the highest value in the vector signals that not
-        ///     all assets should be deployed. For example, `amount_outcome_assets = [120, 150]
-        ///     means, that after deployment 30 of the first outcome asset will be kept.
+        /// * `market_id`: The id of the market.
+        /// * `amount`: The amount of each token to add to the pool.
         /// * `weights`: The relative denormalized weight of each asset price.
         #[pallet::weight(
-            T::WeightInfo::buy_complete_set(T::MaxCategories::get().min(amount_outcome_assets.len().saturated_into()).into())
-            .saturating_add(T::WeightInfo::deploy_swap_pool_for_market(T::MaxCategories::get().min(weights.len().saturated_into()).into()))
+            T::WeightInfo::buy_complete_set(T::MaxCategories::get().into())
+            .saturating_add(T::WeightInfo::deploy_swap_pool_for_market(
+                T::MaxCategories::get().into(),
+            ))
             // Overly generous estimation, since we have no access to Swaps WeightInfo
             // (it is loosely coupled to this pallet using a trait). Contains weight for
             // create_pool() and swap_exact_amount_in()
-            .saturating_add(5_000_000_000.saturating_mul(T::MaxCategories::get().min(amount_outcome_assets.len().saturated_into()).into()))
+            .saturating_add(5_000_000_000.saturating_mul(T::MaxCategories::get().into()))
             .saturating_add(T::DbWeight::get().reads(2 as Weight))
         )]
         #[transactional]
         pub fn deploy_swap_pool_and_additional_liquidity(
```

Comment:
Technically speaking this function is not necessary anymore, because anyone can use `pallet-utility` to batch the `buy_complete_set` and `deploy_swap_pool_for_market` functions into one transaction. However, I am fine with keeping it since it's more straightforward and we already have it.

Reply:
Let's keep it for now. We can remove it at some point down the road.